app/src/main/kotlin/de/janhopp/luebeckmensawidget/WidgetConfigurationActivity.kt

```diff
@@ -0,0 +1,39 @@
+package de.janhopp.luebeckmensawidget
+
+import android.appwidget.AppWidgetManager.EXTRA_APPWIDGET_ID
+import android.appwidget.AppWidgetManager.INVALID_APPWIDGET_ID
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.enableEdgeToEdge
+import de.janhopp.luebeckmensawidget.ui.config.WidgetConfigurationScreen
+import de.janhopp.luebeckmensawidget.ui.theme.MensaTheme
+
+class WidgetConfigurationActivity : ComponentActivity() {
+    private var appWidgetId = INVALID_APPWIDGET_ID
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+
+        appWidgetId = intent?.extras?.getInt(EXTRA_APPWIDGET_ID, INVALID_APPWIDGET_ID)
+            ?: INVALID_APPWIDGET_ID
+
+        setContent {
+            MensaTheme {
+                WidgetConfigurationScreen(
+                    onConfigurationFinished = {
+                        setResult(RESULT_OK, Intent().putExtra(EXTRA_APPWIDGET_ID, appWidgetId))
+                        finish()
```

Comment:
Is there some way to trigger a reload of the widget here? On my device, I have to manually click the update button after I made changes.

Reply:
There is for old widgets, but I couldn't figure it out for glance widgets...
Maybe we make an issue and figure it out later

Reply:
This looks promising: https://stackoverflow.com/questions/70502137/how-to-update-the-appwidgetstate-of-a-compose-glance-widget-from-a-configuration

Reply:
I have found a shorter solution that should also work just as well for now

Reply:
The second time sometimes does not work as quickly though...

Reply:
That doesn't work on my machine 😱 In the logs it says that the widget is updated, but the configuration changes are not applied.